win_setupgui/src/gui.rs

```diff
@@ -0,0 +1,278 @@
+extern crate native_windows_derive as nwd;
+extern crate native_windows_gui as nwg;
+
+use mongo_odbc_core::util::dsn::DSNOpts;
+use nwd::NwgUi;
+use nwg::NativeUi;
+use std::{cell::RefCell, thread};
+use windows::Win32::System::Search::{ODBC_ADD_DSN, ODBC_CONFIG_DSN};
+
+#[derive(Default, NwgUi)]
+pub struct ConfigGui {
+    dialog_data: RefCell<Option<thread::JoinHandle<String>>>,
+
+    #[nwg_control(size: (600, 450), position: (500, 500), title: "MongoDB Atlas SQL ODBC Driver DSN Configuration", flags: "WINDOW|VISIBLE")]
+    #[nwg_events( OnWindowClose: [ConfigGui::close_cancel] )]
+    window: nwg::Window,
+
+    #[nwg_layout(parent: window, spacing: 5)]
+    grid: nwg::GridLayout,
+
+    #[nwg_control(flags: "VISIBLE", text: "DSN")]
+    #[nwg_layout_item(layout: grid, row: 1, col: 0, col_span: 2)]
+    dsn_field: nwg::Label,
+
+    #[nwg_control(flags: "VISIBLE", text: "", focus: true)]
+    #[nwg_layout_item(layout: grid, row: 1, col: 2, col_span: 7)]
+    dsn_input: nwg::TextInput,
+
+    #[nwg_control(flags: "VISIBLE", text: "Username")]
+    #[nwg_layout_item(layout: grid,  row: 2, col: 0, col_span: 2)]
+    user_field: nwg::Label,
+
+    #[nwg_control(flags: "VISIBLE", text: "")]
+    #[nwg_layout_item(layout: grid,  row: 2, col: 2, col_span: 7)]
+    user_input: nwg::TextInput,
+
+    #[nwg_control(flags: "VISIBLE", text: "Password")]
+    #[nwg_layout_item(layout: grid,  row: 3, col: 0, col_span: 2)]
+    password_field: nwg::Label,
+
+    #[nwg_control(flags: "VISIBLE", text: "", password: Some('*'))]
+    #[nwg_layout_item(layout: grid,  row: 3, col: 2, col_span: 7)]
+    password_input: nwg::TextInput,
+
+    #[nwg_control(flags: "VISIBLE", text: "MongoDB URI")]
+    #[nwg_layout_item(layout: grid,  row: 4, col: 0, col_span: 2)]
+    mongodb_uri_field: nwg::Label,
+
+    #[nwg_control(flags: "VISIBLE", text: "")]
+    #[nwg_layout_item(layout: grid,  row: 4, col: 2, col_span: 7)]
+    mongodb_uri_input: nwg::TextInput,
+
+    #[nwg_control(flags: "VISIBLE", text: "Database")]
+    #[nwg_layout_item(layout: grid,  row: 5, col: 0, col_span: 2)]
+    database_field: nwg::Label,
+
+    #[nwg_control(flags: "VISIBLE", text: "")]
+    #[nwg_layout_item(layout: grid,  row: 5, col: 2, col_span: 7)]
+    database_input: nwg::TextInput,
+
+    // #[nwg_control(flags: "VISIBLE", text: "Log Path")]
+    // #[nwg_layout_item(layout: grid,  row: 6, col: 0, col_span: 2)]
+    // log_path_field: nwg::Label,
+
+    // #[nwg_resource(title: "Select Directory", action: nwg::FileDialogAction::OpenDirectory)]
+    // dialog: nwg::FileDialog,
+
+    // #[nwg_control(flags: "VISIBLE")]
+    // #[nwg_layout_item(layout: grid,  row: 6, col: 2, col_span: 1)]
+    // #[nwg_events( OnButtonClick: [ConfigGui::open_dir_picker] )]
+    // directory_button: nwg::Button,
+
```

Comment:
Some of these commented out fields appear to be the same. Let's get rid of the duplicates.

Reply:
I looked again and I didn't see any duplicates of the same name. The button in this line is required to actually pop the dialog open.

Reply:
Ah, ok